components/camera/camera_test.go

```diff
@@ -168,35 +169,46 @@ func (cs *cloudSource) NextPointCloud(ctx context.Context) (pointcloud.PointClou
 }
 
 func TestCameraWithNoProjector(t *testing.T) {
+	logger := logging.NewTestLogger(t)
 	videoSrc := &simpleSource{"rimage/board1"}
 	noProj, err := camera.NewVideoSourceFromReader(context.Background(), videoSrc, nil, camera.DepthStream)
 	test.That(t, err, test.ShouldBeNil)
 	_, err = noProj.NextPointCloud(context.Background())
 	test.That(t, errors.Is(err, transform.ErrNoIntrinsics), test.ShouldBeTrue)
 
 	// make a camera with a NextPointCloudFunction
-	videoSrc2 := &cloudSource{Named: camera.Named("foo").AsNamed(), simpleSource: videoSrc}
-	noProj2, err := camera.NewVideoSourceFromReader(context.Background(), videoSrc2, nil, camera.DepthStream)
+	cloudSrc2 := &cloudSource{Named: camera.Named("foo").AsNamed(), simpleSource: videoSrc}
+	videoSrc2, err := camera.NewVideoSourceFromReader(context.Background(), cloudSrc2, nil, camera.DepthStream)
+	noProj2 := camera.FromVideoSource(resource.NewName(camera.API, "bar"), videoSrc2, logger)
 	test.That(t, err, test.ShouldBeNil)
 	pc, err := noProj2.NextPointCloud(context.Background())
 	test.That(t, err, test.ShouldBeNil)
 	_, got := pc.At(0, 0, 0)
 	test.That(t, got, test.ShouldBeTrue)
 
-	img, _, err := camera.ReadImage(
-		gostream.WithMIMETypeHint(context.Background(), rutils.WithLazyMIMEType(rutils.MimeTypePNG)),
-		noProj2)
+	// TODO(hexbabe): remove below test when Stream is refactored
+	t.Run("ReadImage depth map without projector", func(t *testing.T) {
+		img, _, err := camera.ReadImage(
+			gostream.WithMIMETypeHint(context.Background(), rutils.WithLazyMIMEType(rutils.MimeTypePNG)),
+			noProj2)
+		test.That(t, err, test.ShouldBeNil)
+		depthImg := img.(*rimage.DepthMap)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, depthImg.Bounds().Dx(), test.ShouldEqual, 1280)
+		test.That(t, depthImg.Bounds().Dy(), test.ShouldEqual, 720)
+	})
+
+	img, err := camera.DecodeImageFromCamera(context.Background(), rutils.WithLazyMIMEType(rutils.MimeTypePNG), nil, noProj2)
```

Comment:
From what I understand atm, it shouldn't since `DecodeImageFromCamera` returns a value from `rimage.DecodeImage`, which returns `image.Image` not `rimage.DepthMap`.

Reply:
image.Image is an interface type: https://cs.opensource.google/go/go/+/refs/tags/go1.23.3:src/image/image.go;l=52 not a concrete type. What is the concrete type?

Reply:
Added `t.Logf("img type: %T", img)` and saw 

`camera_test.go:202: img type: *rimage.LazyEncodedImage`

Reply:
this is fine